### PR TITLE
Mic 4667/metadata file

### PR DESCRIPTION
**Pull request by albrja**
### Adds feature to write out metadata file in post-processing to port over values to pseudopeople
- *Category*: Post-processing
- *JIRA issue*: [MIC-4667](https://jira.ihme.washington.edu/browse/MIC-4667)
- *Research reference*: 

### Changes and notes
-adds metadata file to post-processing

### Verification and Testing
Ran post-processing and metadata file is written

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/tools/cli.py [hunk 1]
```diff
@@ -66,14 +66,9 @@ def make_artifacts(
...
-    "-V",
-    "--label-version",
-    type=str,
-    default=None,
-    show_default=True,
-    help="Provide a version number for final results. "
-    "Version should follow format of '#.#.#'.",
+@click.argument(
+    "label_version",
+    # Version for final results and metadata file. This should be of format '#.#.#'
```
Comment (mattkappel) [inline src/vivarium_census_prl_synth_pop/tools/cli.py:71]: This isn't enforced. Should it?

Comment (stevebachmeier) [reply src/vivarium_census_prl_synth_pop/tools/cli.py:71]: Yes it is https://github.com/ihmeuw/vivarium_census_prl_synth_pop/blob/main/src/vivarium_census_prl_synth_pop/tools/cli.py#L184

File: src/vivarium_census_prl_synth_pop/tools/cli.py [hunk 2]
```diff
@@ -66,14 +66,9 @@ def make_artifacts(
...
 @click.command()
 @click.argument("output_dir", type=click.Path(exists=True))
-@click.option(
-    "-V",
-    "--label-version",
-    type=str,
-    default=None,
-    show_default=True,
-    help="Provide a version number for final results. "
-    "Version should follow format of '#.#.#'.",
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/tools/cli.py:76]: Why did you remove this help text?

File: src/vivarium_census_prl_synth_pop/utilities.py [hunk 3]
```diff
@@ -532,3 +532,10 @@ def copy_from_household_member(
             )
 
     return pop
+
+
+def write_metadata_file(final_output_dir: Path, label_version: str) -> None:
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/utilities.py:537]: Why is this a parquet file? Can't it just be a yaml or json file?

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_census_prl_synth_pop/utilities.py [hunk 4]
```diff
@@ -532,3 +532,10 @@ def copy_from_household_member(
...
+    data_version = pd.Series(label_version)
+    metadata = pd.DataFrame({"data_version": data_version})
+    outpath = final_output_dir / "metadata.parquet"
+    metadata.to_parquet(outpath)
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/utilities.py:541]: Do we want the metadata file to be parquet or somethign else like json?

File: src/vivarium_census_prl_synth_pop/utilities.py [hunk 5]
```diff
@@ -532,3 +532,10 @@ def copy_from_household_member(
             )
 
     return pop
+
+
+def write_metadata_file(final_output_dir: Path, label_version: str) -> None:
+    data_version = pd.Series(label_version)
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/utilities.py:538]: NOt sure what the ultimate goal of this metadata file is, but might be worth adding all the things (like all args used as input)

File: src/vivarium_census_prl_synth_pop/tools/cli.py (hunk 2 above)
Comment (stevebachmeier) [reply src/vivarium_census_prl_synth_pop/tools/cli.py:76]: I don't think arguments can have help text. But it could certainly go in the function's docstring

Comment (albrja) [reply src/vivarium_census_prl_synth_pop/tools/cli.py:76]: Arguments can't. I can add a docstring and add it there.

File: src/vivarium_census_prl_synth_pop/utilities.py (hunk 3 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/utilities.py:537]: I was trying to align with our other outputs from post-processing that are either hdf or parquet. We don't know all of what will be going into this file and what the format should be so I was just trying to stay consistent.

File: src/vivarium_census_prl_synth_pop/utilities.py (hunk 4 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/utilities.py:541]: Same response as above - It can be whatever we want it to be. I was trying to align with our other outputs from post-processing that are either hdf or parquet. We don't know all of what will be going into this file and what the format should be so I was just trying to stay consistent.

File: src/vivarium_census_prl_synth_pop/tools/cli.py [hunk 6]
```diff
@@ -66,14 +66,9 @@ def make_artifacts(
...
+@click.argument(
```
[the remark this answers is not in the document]
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/tools/cli.py:69]: I originally had the same thought. My opinion is we have to (at any point) determine a version number to put in the metadata file. My reason for making it a required arg is since we need to it to the metadata file, we should be passing something thats in a consistent format. In pseudopeople we would then have to compare data versions for 2.0.0 or for SOME_RANDOM_DATE..

File: src/vivarium_census_prl_synth_pop/utilities.py (hunk 5 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/utilities.py:538]: We haven't fleshed out everything that will need to be in this file. THe main purpose at some point is going to be to calculate different proportions of the datasets to use for noising in pseudopeople and not have to calculate these values in pseudopeople and do it in post-processing. I'm not sure how helpful run params would be but I'll think about it because adding more stuff now wouldn't hurt.

Comment (rmudambi) [reply src/vivarium_census_prl_synth_pop/utilities.py:538]: This is just to create the metadata file and validate the data version using it for now. We will almost certainly add more things to it.

Comment (NathanielBlairStahn) [reply src/vivarium_census_prl_synth_pop/utilities.py:538]: I don't remember everything we've talked about putting in the metadata file, but it seems like it would be good to think through it and come up with a list of ideas. I think a lot of the use cases came up in the context of providing useful user warnings without having to read the whole dataset. Here are the ones I remember:

- Available years for each dataset
- Available states for each dataset
- What locale the data is for (Anytown, RI, or USA)
- Version number of the data
- Fraction of eligible rows for various noise types (nicknames, copy from household member, guardian-based duplication) -- this may be less relevant if @zmbc [implements concatenating the data before adding noise](https://ihme.slack.com/archives/C02KUQ9LX32/p1698265517189139) rather than noising each shard separately, though we'd still need it for noising in parallel, e.g., using Modin
- I think there was some other issue with nicknames that we talked about potentially putting in a metadata file, but it would require storing a lot of data. I forget exactly what it was -- does anyone remember?

Can anyone think of other things that should go in the metadata file?

**Review by rmudambi — APPROVED** [review]
